stories/ui/Button.js

```diff
@@ -1,185 +1,82 @@
-      {label}
-    </Button>
-  ))
-  .add('action buttonType, puffy, disabled', () => (
-    <Button buttonType="action" puffy disabled>
-      {label}
-    </Button>
-  ))
-  .add('action buttonType, micro', () => (
-    <Button buttonType="action" micro>
-      {label}
-    </Button>
-  ))
-  .add('action buttonType, micro, disabled', () => (
-    <Button buttonType="action" micro disabled>
-      {label}
-    </Button>
-  ))
-  .add('cancel buttonType', () => <Button buttonType="cancel">{label}</Button>)
-  .add('cancel buttonType, disabled', () => (
-    <Button buttonType="cancel" disabled>
-      {label}
-    </Button>
-  ))
-  .add('cancel buttonType, puffy', () => (
-    <Button buttonType="cancel" puffy>
-      {label}
-    </Button>
-  ))
-  .add('cancel buttonType, puffy, disabled', () => (
-    <Button buttonType="cancel" puffy disabled>
-      {label}
-    </Button>
-  ))
-  .add('cancel buttonType, micro', () => (
-    <Button buttonType="cancel" micro>
-      {label}
-    </Button>
-  ))
-  .add('cancel buttonType, micro, disabled', () => (
-    <Button buttonType="cancel" micro disabled>
-      {label}
-    </Button>
-  ))
-  .add('confirm buttonType', () => (
-    <Button buttonType="confirm">{label}</Button>
-  ))
-  .add('confirm buttonType, disabled', () => (
-    <Button buttonType="confirm" disabled>
-      {label}
-    </Button>
-  ))
-  .add('confirm buttonType, puffy', () => (
-    <Button buttonType="confirm" puffy>
-      {label}
-    </Button>
-  ))
-  .add('confirm buttonType, puffy, disabled', () => (
-    <Button buttonType="confirm" puffy disabled>
-      {label}
-    </Button>
-  ))
-  .add('confirm buttonType, micro', () => (
-    <Button buttonType="confirm" micro>
-      {label}
-    </Button>
-  ))
-  .add('confirm buttonType, micro, disabled', () => (
-    <Button buttonType="confirm" micro disabled>
-      {label}
-    </Button>
-  ));
+function createSections(buttonType) {
+  return [
+    {
+      description: `${buttonType || 'default (no buttonType)'}`,
+      props: {
+        buttonType,
+      },
+    },
+    {
+      description: 'disabled=true',
+      props: {
+        buttonType,
+        disabled: true,
+      },
+    },
+    {
+      description: 'puffy=true',
+      props: {
+        buttonType,
+        puffy: true,
+      },
+    },
+    {
+      description: 'puffy=true, disabled=true',
+      props: {
+        buttonType,
+        puffy: true,
+        disabled: true,
+      },
+    },
+    {
+      description: 'micro=true',
+      props: {
+        buttonType,
+        micro: true,
+      },
+    },
+    {
+      description: 'micro=true, disabled=true',
+      props: {
+        buttonType,
+        micro: true,
+        disabled: true,
+      },
+    },
+  ];
+}
+
+function createChapters() {
+  return buttonTypes.map((type) => {
+    return {
+      title: type,
+      sections: createSections(type).map((section) => {
+        return {
+          subtitle: `${section.description}`,
```

Comment:
I like the style of showing a string of props and values but manually defining this as `description` is going to be hard to maintain. I say remove `description` from `section` and just render the same thing by inspecting `section.props`. If you make it a common function then `Badge` can use it too.

Reply:
yep I agree and like your suggestion. I will update..

Reply:
this part is WIP..